### DISCUSSION TITLE
Has anybody a good solution to manage the (etherscan verification) API-keys depending on the chain ?

### DISCUSSION THREAD

**Discussion by SvenMeyer** [PaulRBerg/hardhat-template#69] [General]
the API-key for verification on Etherscan , BSCscan , ... is still hard coded in my `hardhat.config.ts`
Ideally it would be set depending on the provided network ... and suggestions ?

```typescript
  etherscan: {
    apiKey: "42WH....................FK43H92W5N" // process.env.ETHERSCAN_API_KEY,
  },
```

**Comment by SvenMeyer** [comment]
also something similar would also be great to set different mnemonic based on chain, ..

**Comment by PaulRBerg** [comment]
This is an issue with `hardhat-etherscan` itself. It's being tracked here:

https://github.com/nomiclabs/hardhat/issues/1448
